Codec/Archive/Zip.hs

```diff
@@ -327,9 +327,17 @@ sourceEntry s sink = do
 
 saveEntry
   :: EntrySelector     -- ^ Selector that identifies archive entry
+  -> Bool              -- ^ Whether to restore modification time
   -> Path b File       -- ^ Where to save the file
   -> ZipArchive ()
-saveEntry s path = sourceEntry s (CB.sinkFile (toFilePath path))
+saveEntry s restoreModTime path = do
+  sourceEntry s (CB.sinkFile (toFilePath path))
+  when restoreModTime $ do
+    edM <- getEntryDesc s
+    -- If the sourceEntry above worked, the getEntryDesc should return a Just.
+    case edM of
+      Nothing -> error "Unexpectedly missing entry."
```

Comment:
It's better to write it this way: `med <- getEntryDesc; forM_ med (setModificationTime path . edModTime)`.